sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -212,10 +212,8 @@ def fetch_batch
             spans.shift(@batch_size).map!(&:to_span_data)
           end
 
-          def lock
-            @mutex.synchronize do
-              yield
-            end
+          def lock(&block)
+            @mutex.synchronize(&block)
```

Comment:
Same perf concern here.

Reply:
I opted to change back 9ccbeefdf95ad414cd789712a4fadf4ae1aff8f8